cups/file.c

```diff
@@ -335,6 +335,136 @@ _cupsFileCheckFilter(
 #endif /* !_WIN32 */
 
 
+/*
+ * '_cupsFileGetConfAndComments()' - Get line and comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(cups_file_t *fp,	/* I  - CUPS file */
+                char        *buf,	/* O  - String buffer */
+		size_t      buflen,	/* I  - Size of string buffer */
+                char        **value,	/* O  - Pointer to value */
+		int         *linenum)	/* IO - Current line number */
+{
+  char	*ptr;				/* Pointer into line */
+
+
+ /*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+
+    /*
+     * Remove the inline comment...
+     */
+     if ((ptr = strchr(buf, '#')) != NULL && _cups_isspace(*(ptr-1)))
+     {
+        while(ptr > buf)
+        {
+         // Null-terminate the string after the last non-whitespace
+         if(!_cups_isspace(*(ptr-1)))
+         {
+           *ptr = '\0';
+           break;
+         }
+         ptr--;
+        }
+     }
+
+     /*
+      * Strip leading whitespace...
+      */
+
+      for (ptr = buf; _cups_isspace(*ptr); ptr ++);
+
+      if (ptr > buf)
+        _cups_strcpy(buf, ptr);
+
+       /*
+        * Return the comment if any...
+        */
+        if(buf[0])
+        {
+          if (buf[0] == '#')
+            return buf;
+
+          /*
+           * Otherwise grab any value and return...
+           */
+
+          for (ptr = buf; *ptr; ptr ++)
+            if (_cups_isspace(*ptr))
+              break;
+
+          if (*ptr)
+          {
+           /*
+            * Have a value, skip any other spaces...
+            */
+
+            while (_cups_isspace(*ptr))
+              *ptr++ = '\0';
+
+            if (*ptr)
+              *value = ptr;
+
+           /*
+            * Strip trailing whitespace and > for lines that begin with <...
+            */
+
+            ptr += strlen(ptr) - 1;
+
+            if (buf[0] == '<' && *ptr == '>')
+              *ptr-- = '\0';
+            else if (buf[0] == '<' && *ptr != '>')
+            {
+             /*
+              * Syntax error...
+              */
+
+              *value = NULL;
+              return (buf);
+            }
+
+            while (ptr > *value && _cups_isspace(*ptr))
+              *ptr-- = '\0';
+          }
+
+          /*
+           * Return the line...
+           */
+
+           return (buf);
+
+        }
+        return (buf);
```

Comment:
I'm sorry, I didn't realize cupsFileGets() removes the end of lines, so the ```if(buf[0])``` is not met.

Let's ignore the blank lines then - we would have to deal with them by a different cupsFilePuts(), which is not worthy outcome (because we would have to ignore the indent later) - so just remove the return here.

Reply:
Fixed now - we now support empty lines too.